### PR TITLE
WOfS Notebook and installation script for the Hackathon

### LINKED ISSUE #8: Create a WOfS notebook
Create a WOfS demo notebook based on our previous work around Lake Tempe. The WOfS algorithm are pulled from GA's repositories and the DEM from `lpdaac` data. An installation script is required to install dependencies and should be made as seamless to use as possible for the Hackathon participants. An HTML version should also be provided, as usual.

### PR DESCRIPTION
**Pull request by rtaib**
This PR adds a notebook (labelled 03 because 02 is expected to be TSS) and corresponding installation script for its dependencies. It showcases GA's WOfS algorithm and the use of `lpdaac` DEM data.

Fixes #8 

This branch should be deleted after merging.

### CODE REVIEW

**Comment by mpaget** [comment]
One suggestion: Can we rename `03-water-observations-from-space.ipynb` to `case-studies/water-observations-from-space.ipynb` ?

- The `case-studies` directory doesn't exist yet (this would be the first notebook).
- There may be implications (relative paths) for the pre-commit hook that need to be updated.

**Comment by rtaib** [comment]
> One suggestion: Can we rename `03-water-observations-from-space.ipynb` to `case-studies/water-observations-from-space.ipynb` ?
> 
> * The `case-studies` directory doesn't exist yet (this would be the first notebook).
> * There may be implications (relative paths) for the pre-commit hook that need to be updated.

Done in Commit 9849e32

File: case-studies/water-observations-from-space.ipynb [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9f170afa-d44e-4450-a62b-e7400ee87d77",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "summary.frequency.plot(size=10)\n",
+    "plt.title(\"Lake Tempe – Wet frequency for 2020\");"
```
Comment (mpaget) [inline case-studies/water-observations-from-space.ipynb:486]:
```suggestion
    "plt.title(f\"{aoi_label} - Wet frequency for {time[0]} to {time[1]}\");"
```

File: case-studies/water-observations-from-space.ipynb [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "848734be-d09e-4f0f-b0e5-905086e855d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Save frequencies to high-res image file\n",
+    "summary.frequency.plot(size=20)\n",
+    "plt.title(\"Lake Tempe – Wet frequency for 2020\")\n",
```
Comment (mpaget) [inline case-studies/water-observations-from-space.ipynb:498]:
```suggestion
    "plt.title(f\"{aoi_label} - Wet frequency for {time[0]} to {time[1]}\")\n",
```

**Review by mpaget — COMMENTED** [review]
LGTM, with consideration for the suggestions I've made.

File: tools/install_wofs.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        git clone --depth=1 https://csiro-easi@dev.azure.com/csiro-easi/easi-hub-public/_git/hub-notebooks ${repos_dir}/hub-notebooks
+    fi
+}
+
+if [ -d ${venv_root}/${venv} ]; then
+    echo "${venv_name} kernel already installed"
+else
+    create_venv
+    install_dependencies
+    clone_repos
```
[the remark this answers is not in the document]
Comment (rtaib) [reply tools/install_wofs.sh:46]: I removed the whole function.

File: tools/install_wofs.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    deactivate
+}
+
+function clone_repos() {
+    # Clore required repos
+    echo "Cloning additional repos. Please wait..."
+    if [ ! -d ${repos_dir}/hub-notebooks ] ; then
+        git clone --depth=1 https://csiro-easi@dev.azure.com/csiro-easi/easi-hub-public/_git/hub-notebooks ${repos_dir}/hub-notebooks
+    fi
+}
```
[the remark this answers is not in the document]
Comment (rtaib) [reply tools/install_wofs.sh:39]: I removed the whole function.

**Review by mpaget — APPROVED** [review]
Thank you for the updates.

**Pull request merged**